### DISCUSSION TITLE
Commits are not shown in my activity graph

### DISCUSSION THREAD

**Discussion by Kirikosha** [community/community#126960] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello. I am quite new to GitHub and I encountered a very strange problem. I have got 2 github accounts, one for university and one for myself. I have connected them to github through ssh keys, so I can access both of them from my terminal while I am doing commits or whatever. And everything seemed fine, but recently I noticed, that all commits that are made from the account I am writing this question (Kirikosha) aren't being shown in my Activity. 
![firefox_2Ec4cNpodw](https://github.com/community/community/assets/109356230/25219a72-7dac-4003-abd1-6abde4377372)

I have checked a lot of thins before:

- My git email is identical to the github email
- I have set the mode to show activity even in private repositories

The weirdest thing probably is that all commits are showing my account as an author, but they aren't shown in my activity graph:
![image](https://github.com/community/community/assets/109356230/5586ac4b-eb0c-40c8-9a82-6ce8807a0755)

I hope you may help me with this problem, thank you for your future response and support

**Comment by davevad93** [comment]
Hi @Kirikosha ,

Usually when contributions aren't shown in the activity graph it's for the following reasons:

1. Email not properly configured in `git config`.
2. Committing to a secondary `branch`. 
3. `Include private contributions in my profile` option unchecked in the `Settings`.

As far as I can see that doesn't seem to be your problem. The only thing I can think of is that having 2 different accounts connected may be causing some conflict in the global git configuration. Run `git config --global user.name "Kirikosha"`, make a `commit` and check if it fixes your issue.

**Reply by NureBiletskyiDanylo** [reply]
Thank you very much for your reply. May I ask you one more question? If I make a commit from one account to another account's repository as a contributor, should it appear in my activity graph? Thank you for your future response

**Reply by ujjwaljha1** [reply]
Yes

> Thank you very much for your reply. May I ask you one more question? If I make a commit from one account to another account's repository as a contributor, should it appear in my activity graph? Thank you for your future response

YeS

**Reply by Kirikosha** [reply]
Well, then it is very strange, because if I make a commit as the author of a repository, they are shown. But if I make a commit as a contributor to another account, it isn't shown. Is it possible that a thing called `git remote add origin` plays a crucial role? When I create a repository for one account or another, I always add this remote origin to the email of the author account through ssh (for example, `git remote add origin git@uni.github.com...` (for university account)

**Reply by davevad93** [reply]
Hey @Kirikosha you're welcome, yes if you make a `commit` in the main `branch` of another account's repo it will appear in your activity graph as a contribution.  If you make a `commit` on a secondary `branch`, it will only count as a contribution when merged into the main one and therefore it will count as 2 contributions, your `commit` and the merged `pull request`. If you need more context on this subject read this [comment](https://github.com/orgs/community/discussions/125840#discussioncomment-9582812) from an old discussion.

**Reply by davevad93** [reply, marked as answer]
Sorry @Kirikosha I didn't see your last comment, to be sure try with the following:

1. Clone your repo running `git clone git@uni.github.com...`.
2. Make some changes and add them to the staging are with `git add .`.
3. Commit your changes with `git commit`.
4. Push the changes to your remote repo running `git push origin main` or `master` depending on your default branch name.

If by doing this your contributions appear, it's very likely that the issue is due to what you have just mentioned.

**Reply by Kirikosha** [reply]
Thank you very much for your reply. I have solved this problem, but it wasn't exactly the way you told me. It appears that if you use the ssh key and make commits using it from one account to another, GitHub will add you as a "ghost" contributor. In fact, I was making commits, but I wasn't a real contributor. So I deleted all ssh keys and made a reset for my git config. After that, I noticed that I was not in contributors because I had no access to push. After that, I added myself as a contributor, and it worked. I can see commits in my github now. Thank you very much for your help and support. Have a nice day!!!

**Reply by davevad93** [reply]
You're welcome @Kirikosha  I'm happy you finally solved it, have a nice day too!
